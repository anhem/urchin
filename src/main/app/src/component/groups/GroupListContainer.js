import React, {Component} from 'react'
import {connect} from 'react-redux'
import {getGroups} from '../../action/groupAction'
import GroupList from "./GroupList";

class GroupListContainer extends Component {

    componentWillMount() {
        this.props.getGroups();
    }

    render() {
        return (
            <GroupList
                groups={this.props.groups}
            />
        )
    }
}

const mapDispatchToProps = (dispatch) => {
    return {
        getGroups: () => {
            dispatch(getGroups())
        }
    }
};

const mapStateToProps = (state) => {
    return {
        groups: state.groupData.groups || []
    }
};

export default connect(mapStateToProps, mapDispatchToProps)(GroupListContainer)