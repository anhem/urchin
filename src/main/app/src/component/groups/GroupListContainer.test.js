import React from 'react';
import {mount, shallow} from 'enzyme'
import toJson from 'enzyme-to-json';
import {GroupListContainer} from "./GroupListContainer";

describe('GroupListContainer', () => {

    let props = {
        getGroups: jest.fn(),
    };

    it('renders without crashing', () => {
        expect(mount(<GroupListContainer {...props}/>).length).toEqual(1);
    });

    it('match snapshot', () => {
        expect(toJson(shallow(<GroupListContainer {...props}/>))).toMatchSnapshot();
    });
});
