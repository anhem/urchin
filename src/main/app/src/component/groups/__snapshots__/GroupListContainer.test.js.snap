// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GroupListContainer match snapshot 1`] = `<GroupList />`;
